OutOfSchool/OutOfSchool.WebApi/Controllers/V1/CompetitiveEventController.cs

```diff
@@ -144,6 +145,39 @@ public async Task<IActionResult> Delete(Guid id)
         return NoContent();
     }
 
+    /// <summary>
+    /// Retrieves a list of competitive event view cards for a specific provider.
+    /// </summary>
+    /// <param name="id">
+    /// The unique identifier of the provider. This value must not be an empty GUID.
+    /// </param>
+    /// <param name="filter">
+    /// Filter to get specified portion of competitive events' view cards for specified provider, Id of the excluded workshop could be specified.
+    /// If <see cref="ExcludeIdFilter.ExcludedId"/> is provided, it must not be an empty GUID.
+    /// </param>
+    /// <returns>
+    /// <see cref="SearchResult{CompetitiveEventViewCardDto}"/>, or no content
+    /// </returns>
+    [AllowAnonymous]
+    [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(SearchResult<CompetitiveEventViewCardDto>))]
+    [ProducesResponseType(StatusCodes.Status204NoContent)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    [HttpGet("/api/v{version:apiVersion}/provider/{id}/competitiveevents")]
+    public async Task<IActionResult> GetCompetitiveEventViewCardByProviderId(Guid id, [FromQuery] ExcludeIdFilter filter)
+    {
+        if (id == Guid.Empty)
+        {
+            return BadRequest("Provider id is empty.");
+        }
+        if (filter?.ExcludedId == Guid.Empty)
```

Comment:
don't do this please, this is not a bad request. You already have a check in service, just extend it, e.g., `filter.ExcludedId is not null && filter.ExcludedId != Guid.Empty`